builder/dockerfile/shell_parser.go

```diff
@@ -169,25 +170,23 @@ func (sw *shellWord) processSingleQuote() (string, error) {
 	//   all the characters (except single quotes, making it impossible to put
 	//   single-quotes in a single-quoted string).
 
-	var result string
+	var result bytes.Buffer
 
 	sw.scanner.Next()
 
+loop:
```

Comment:
Yes, it rubs me a bit the wrong way as well, but I think it might be an unfortunate necessity in golang.

While it looks like a GOTO, I don't think it acts like one. It still acts like a regular break statement, it just gives you a way to target the "thing" you're breaking out of.

An alternative is to just return here instead. Nothing is done after this break, so no other code changes are required. I went back and forth between these options and thought I'd try this first.

I'm fine to change this to an early return.